x/vesting/keeper/msg_server.go

```diff
@@ -215,10 +215,19 @@ func (k Keeper) Clawback(
 		dest = funder
 	}
 
-	if k.authority.String() != msg.FunderAddress && bk.BlockedAddr(dest) {
-		return nil, errorsmod.Wrapf(errortypes.ErrUnauthorized,
-			"%s is a blocked address and not allowed to receive funds", msg.DestAddress,
-		)
+	if msg.FunderAddress != k.authority.String() {
+		if k.HasActiveClawbackProposal(ctx, addr) {
+			return nil, errorsmod.Wrapf(errortypes.ErrUnauthorized,
+				"clawback is disabled while there is an active clawback proposal for account %s and funder %s",
+				msg.AccountAddress, msg.FunderAddress,
+			)
+		}
+
+		if bk.BlockedAddr(dest) {
+			return nil, errorsmod.Wrapf(errortypes.ErrUnauthorized,
+				"%s is a blocked address and not allowed to receive funds", msg.DestAddress,
+			)
```

Comment:
I'm wondering if we should enforce the blocked addresses check for the destination regardless if it through gov or not?

Reply:
yes definitely, good catch! although when the clawback is initiated by the module address the proposal must have already passed so the check should already be done when submitting the proposal 👍

Reply:
ah I noticed that this check does make sense only for this specific case, because we are hardcoding the destination address to be the community pool for the other case anyways:
https://github.com/evmos/evmos/blob/9776489096c0e62538a64ead383e4191b8e6de2d/x/vesting/keeper/msg_server.go#L248-L253